### PR TITLE
Bug fix for test_range and test_zoom_level

### PR DESCRIPTION
**Pull request by hanyelgaml-roche**

### CODE REVIEW

File: src/efs_parser/BigWig.py [hunk 1]
```diff
@@ -184,7 +184,15 @@ def getRange(self, chr, start, end, bins=2000, zoomlvl=-1, metric="AVG", respTyp
             if respType == "DataFrame":
                 result = toDataFrame(values, self.columns)
                 result["chr"] = chr
+                # checking range by: Hany Elgaml
 
+                query_string = '(start >= ' + str(start) + ' and start <= ' + str(end) + ') or (end >= ' + str(
+                    start) + ' and end <= ' + str(end) + ')'
```
Comment (jkanche) [inline src/efs_parser/BigWig.py]: we can simplify this ever further 

```
query_string = f"start <= {end} and end >= {start}" 
```

also checkout f strings in python

File: src/efs_parser/BigWig.py [hunk 2]
```diff
@@ -184,7 +184,15 @@ def getRange(self, chr, start, end, bins=2000, zoomlvl=-1, metric="AVG", respTyp
...
+                result = result.query(query_string)
+                result.insert(1, "new", range(0, len(result)))
+                result.set_index('new', inplace=True)
```
Comment (jkanche) [inline src/efs_parser/BigWig.py]: we don't need to reindex the result, so lets remove these lines

File: tests/test_BigBed_remote.py [hunk 3]
```diff
@@ -31,8 +31,8 @@ def test_range():
     end = 10020000
     res, err = bb.getRange(chr="chr1", start=start, end=end)
     assert(err == None)
-    assert(len(res) == 437)
-    for i in range(0,437):
+    assert(len(res) == 102)
+    for i in range(0,102):
```
Comment (jkanche) [inline tests/test_BigBed_remote.py]: pandas provides a method `iterrows` to iterate through the dataframe. That might be more appropriate here. 

I would also avoid checking explicit lengths

Comment (jkanche) [reply tests/test_BigBed_remote.py]: the `test_range` for BigWigs also need to check if the result is within the range we query

**Review by jkanche — CHANGES_REQUESTED** [review]

**Comment by jkanche** [comment]
Also, your commit includes your entire `venv` directory. Please make sure to remove this in another commit

**Comment by jkanche** [comment]
Looking at why the build is failing, we have a couple of more issues - 

For tests with `TranscriptTbx` and `GTFParsed`, use the following urls instead of file paths - 

- transcripts - https://raw.github.com/epiviz/efs-genomes/master/hg38/transcripts.tsv.bgz
- genes - https://raw.github.com/epiviz/efs-genomes/master/hg38/genes.tsv.gz

**Comment by jkanche** [comment]
there's duplication of commits in the PR.

I suggest re-forking the repository and making these changes again (without the `venv` folder), 
(1) you can either hard reset your `main` to `epiviz/main` or (2) pull in `epiviz/main` as a new branch into your local repo. either way should work. Probably (2) is easier/better

a few links on git and github

- https://guides.github.com/
- https://medium.com/automationschool/git-super-basic-primer-for-beginner-developers-and-qa-part-1-b62e46863f75
- https://www.atlassian.com/git

File: src/efs_parser/BigWig.py (hunk 1 above)
Comment (hanyelgaml-roche) [reply src/efs_parser/BigWig.py]: Done

File: src/efs_parser/BigWig.py (hunk 2 above)
Comment (hanyelgaml-roche) [reply src/efs_parser/BigWig.py]: done

File: tests/test_BigBed_remote.py (hunk 3 above)
Comment (hanyelgaml-roche) [reply tests/test_BigBed_remote.py]: done

**Pull request closed without merging**